Lib/test/support/__init__.py

```diff
@@ -1982,3 +1983,13 @@ def skip_if_broken_multiprocessing_synchronize():
             synchronize.Lock(ctx=None)
         except OSError as exc:
             raise unittest.SkipTest(f"broken multiprocessing SemLock: {exc!r}")
+
+
+def check_disallow_instantiation(testcase, tp, *args, **kwds):
```

Comment:
Maybe rename "tp" to "type" or "test_type".

Reply:
I prefer *tp* or *test_type*. For the latter, I'd also rename the first argument to *test_cast* for consistency. But I think *tp* is fine. It's a common abbreviation for *type* in the CPython code base.